llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -6858,8 +6858,9 @@ SDValue SelectionDAG::getNode(unsigned Opcode, const SDLoc &DL, EVT VT,
     // expanding copies of large vectors from registers. This only works for
     // fixed length vectors, since we need to know the exact number of
     // elements.
-    if (N2C && N1.getOperand(0).getValueType().isFixedLengthVector() &&
```

Comment:
Can we do

```
if (N2C && N1.getOpcode() == ISD::CONCAT_VECTORS &&
    N1.getOperand(0).getValueType().isFixedLengthVector())
```

I don't think we need the `N1.getNumOperands() > 0`. A CONCAT_VECTORS should never be created with no operands. It would fail an assert in `foldCONCAT_VECTORS` if we tried to create it.

Reply:
yes, I have changed it.